lib/Zonemaster/Backend/DB.pm

```diff
@@ -130,6 +133,30 @@ sub process_unfinished_tests {
     }
 }
 
+sub generate_deterministic_hash {
+    my ( $self, $params ) = @_;
+
+    my $profile = Zonemaster::Engine::Profile->effective;
+
+    my %to_encode = ();
+    $to_encode{domain}      = lc $$params{domain}   // "";
```

Comment:
1. Is it possible to come here with an undefined "domain"?
2. I am not sure that this is the correct place in the code to do normalization (down casing) of the domain name. Normalization of domain name is a bigger issue with IDN and should reside in a central place such as RPCAPI.pm

Reply:
> Is it possible to come here with an undefined "domain"?

Not sure, I need to investigate.

> I am not sure that this is the correct place in the code to do normalization (down casing) of the domain name.

I totally agree with you. However, we should start somewhere and I think this is a good place before moving it in an upper layer.

Reply:
> Is it possible to come here with an undefined "domain"?

Indeed, this is not possible. Same for `priority` and `queue`. I'll remove the default value.

Reply:
I choose to keep the default value for now as I need it in the unit tests. This is not the best as this duplicate the value. I haven't find a proper solution so far. I'll think about it.